client/app/tasks/components/TasksCalendar.tsx

```diff
@@ -82,31 +85,32 @@ const TasksCalendar = () => {
       `${localizer?.format(start, 'MMM d', culture)} - ${localizer?.format(end, 'MMM d', culture)}`,
   }
 
-  const handleSelectEvent = (event: CustomEventType) => {
-    setSelectedEvent(event)
-
-    // TODO - This is just temp code to show the selected event
-    window.alert('Selected event: ' + event.title)
-  }
-
   if (isLoading || !tasks) return <Loading />
 
   return (
-    <div className='task-calendar h-[600px] md:h-[700px]'>
-      <Calendar
-        localizer={localizer}
-        events={formattedTasks}
-        date={date}
-        onNavigate={handleNavigate}
-        formats={formats}
-        components={{
-          event: CustomEvent,
-          toolbar: CustomToolbar,
-        }}
-        popup={true}
-        onSelectEvent={handleSelectEvent}
+    <>
+      <div className='task-calendar h-[600px] md:h-[700px]'>
+        <Calendar
+          localizer={localizer}
+          events={formattedTasks}
+          date={date}
+          onNavigate={handleNavigate}
+          formats={formats}
+          components={{
+            event: CustomEvent,
+            toolbar: CustomToolbar,
+          }}
+          popup={true}
+          onSelectEvent={handleSelectEvent}
+        />
+      </div>
+      <TaskDetailsModal
```

Comment:
Might this be better so that you don't have to care about null possibility in TaskDetailsModal?

```
{selectedTask && (
        <TaskDetailsModal
          selectedTask={selectedTask}
          onClose={onModalClose}
          onOpen={onModalOpen}
          isModalOpen={isModalOpen}
        />
      )}
```

Reply:
Yeah right now I should have done like this. I fixed it!
https://github.com/HAK2024/fairy-share/pull/114/commits/01e827543c57dfcd337f48dcfb01a7d7d889d858